modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -1010,6 +1019,14 @@ def duty_begin_date_is_after_approximate_begin_date?(begin_date, approximate_beg
         date_regex_groups(begin_date) > date_regex_groups(approximate_begin_date)
       end
 
+      def overlapping_confinement_periods?(idx)
+        @ranges.count > 1 ? date_range_overlap?(@ranges[idx - 1], @ranges[idx]) : return
+      end
+
+      def date_range_overlap?(range_one, range_two)
+        range_one.last > range_two.first || range_two.last < range_one.first
```

Comment:
For approximate dates (YYYY-MM), the logic converts such dates to be the first of the month when it creates a date object from them, so for example 2024-01 is converted to 2024-01-01. This came up in [API-37244](https://jira.devops.va.gov/browse/API-37244), and Alex suggested setting the approximate end date to be the last day of the month. Wondering if we should do that here too?

Reply:
@mchristiansonVA Might be worth having a team discussion on this. When it came up before I think we landed on checking if it was like <= or >= in the cases where we were checking an approximate date against a more exact date. I believe there is previous validation where that is in place and this is the same scenario so should either follow suite or change all the checks. This was the initial [slack thread](https://oddball.slack.com/archives/C019XKGS1CN/p1685713027643999?thread_ts=1685636217.231329&cid=C019XKGS1CN) a long while ago and in the file, for example in the method def date_is_valid_against_current_time_after_check_on_format?(date) on line 1016 we use that <= check.